tests/unit/test-cmd/test.run.js

```diff
@@ -325,5 +326,101 @@ describe('run', () => {
            sourceDir: '/fake/source/dir',
          },
        );
-     });
-});
+     }
+  );
+
+  describe('profile-create-new option', () => {
+    beforeEach(() => {
+      sinon.stub(fs, 'mkdir');
+    });
+
+    afterEach(() => {
+      fs.mkdir.restore();
+    });
+
+    it('creates dir when firefox profile doesn\'t exist',
+       async () => {
+         sinon.stub(fs, 'existsSync').returns(false);
+         const firefoxProfile = '/pretend/path/to/Firefox/profile';
+         const cmd = prepareRun();
+
+         await cmd.run({
+           firefoxProfile,
+           profileCreateIfMissing: true,
+         });
+
+         sinon.assert.calledWith(fs.mkdir, firefoxProfile);
+
+         fs.existsSync.restore();
+       }
+    );
+
+    it('creates dir when chromium profile doesn\'t exist',
+       async () => {
+         sinon.stub(fs, 'existsSync').returns(false);
+         const chromiumProfile = '/pretend/path/to/Chromium/profile';
+         const cmd = prepareRun();
+
+         await cmd.run({
+           chromiumProfile,
+           target: 'chromium',
+           profileCreateIfMissing: true,
+         });
+
+         sinon.assert.calledWith(fs.mkdir, chromiumProfile);
+
+         fs.existsSync.restore();
+       }
+    );
```

Comment:
These two tests are so similar to each other that it may be reasonable to refactor it into a single parametrized test, e.g. something like:

```
async function testCreateProfileIfMissing(runParams, expectedMkdirParam) {
  sinon.stub(fs, 'existsSync').returns(false);
  const cmd = prepareRun();
  await cmd.run({ ...runParams, promiseCreateIfMissing: true });
  sinon.assert.calledWith(fs.mkdir, fs.mkdir, expectedMkdirParam);
}

const fakeProfilePath = '/pretend/profile/path'

it('creates a Firefox profile directory if it does not exist',
   testCreateProfileIfMissing.bind({ firefoxProfile: fakeProfielPath }, fakeProfilePath));

it('creates a Chromium profile directory if it does not exist',
   testCreateProfileIfMissing.bind(
     { chromiumProfile: fakeProfielPath, target: 'chromium' },
     fakeProfilePath));
```

does that sounds reasonable to you too?

Reply:
I definitely agree there is quite a bit of duplication in the tests. I actually feel even the last 2 tests are very similar.
How do you feel about going a step further:

```js
async function testCreateProfileIfMissing(existsSyncReturnValue, runParams, expectedMkdirParam) {
  sinon.stub(fs, 'existsSync').returns(existsSyncReturnValue);
  const cmd = prepareRun();
  await cmd.run({ ...runParams, promiseCreateIfMissing: true });
  sinon.assert.calledWith(fs.mkdir, fs.mkdir, expectedMkdirParam);
}

const fakeProfilePath = '/pretend/profile/path'

it('creates a Firefox profile directory if it does not exist',
   testCreateProfileIfMissing.bind(false, { firefoxProfile: fakeProfielPath }, fakeProfilePath));

it('creates a Chromium profile directory if it does not exist',
   testCreateProfileIfMissing.bind(
     false,
     { chromiumProfile: fakeProfielPath, target: 'chromium' },
     fakeProfilePath));

it('Uses profile directory when firefox profile directory already exists',
   testCreateProfileIfMissing.bind(true, { firefoxProfile: fakeProfielPath }, fakeProfilePath));

it('Uses profile directory when chromium profile directory already exists',
   testCreateProfileIfMissing.bind(
     true,
     { chromiumProfile: fakeProfielPath, target: 'chromium' },
     fakeProfilePath));
```

Reply:
@akhilpanchal sure thing, if all four tests can be expressed with a single parametrized function and still be readable enough (so that we will not have to spent to much time to decode a failure, if any of these test would fail at some point in the future).

I think that for the last 2 test cases the assertion `sinon.assert.calledWith(fs.mkdir, ...)` should actually be replaced by one that assert that mkdir has not been called at all.

(I guess that if we may also assert in all 4 tests that we did run the extension runner with the expected profile dir as the last 2 tests seems to be doing in this version).

Reply:
@rpl 
Thanks for all your feedback 👍.
I addressed all your comments in the latest commit.

I did abstract the common test code into a single function.
Does the implementation look reasonable to you?
If not, I can either split them out to 2 different parameterized tests like you suggested earlier, OR any other way you feel that makes the tests more readable and easier to maintain.
Let me know.